lemmy-almalinux.yml

```diff
@@ -13,6 +13,12 @@
         fail_msg: "This playbook requires Ansible 2.11.0 or higher"
       become: false
 
+    - name: Disable SELinux
+      ansible.posix.selinux:
+        state: disabled
+      tags:
+        - selinux
+
```

Comment:
I'm not sure disabling selinux is the correct (forced?) solution for this. But again, I do not have enough redhat experience for this. So if you think this is the best solution then i'll roll with it.

Reply:
I kept waffling on this, too. It is less than ideal, and I wouldn't say I like it, but to get the playbooks to run cleanly, SELinux needs to be disabled/set to permissive (or modified outside the scope of this playbook). I decided to disable it for the same reason I thought it was okay to punch 80/443 into the firewalls--people expect to run these playbooks cleanly out-of-the-box (see the referenced issue, for example). 

My initial approach was to detect if SELinux is enabled, and if so, prompt the user, warning them it might fail until they disable it. Alternatively, a prominent warning/addition to the README may be sufficient (or a combination). 

Do you have any thoughts on this more passive approach? The user experience is worse, but it’s more “proper” (and secure) to leave SELinux alone and have the user manually turn it off.

Reply:
If you disable selinux then I would at least expect that you disable it at the end of the playbook. And then its still possible that an error is thrown somewhere and it stays disabled which doesnt sound good.

Reply:
> If you disable selinux then I would at least expect that you disable it at the end of the playbook

The playbook & Lemmy deployment will not work with SELinux enabled/enforcing. It must be disabled at the start to ensure clean deployment & functioning. 

> And then its still possible that an error is thrown somewhere and it stays disabled which doesnt sound good.

I think I missed an important reason I took this approach—Lemmy being deployed via these playbooks *will not* function without additional tweaking of SELinux policies and contexts, which I believe is outside the scope of this playbook. 

While it’s not the best, this is why some cloud providers ship RHEL9 images with it disabled by default, even if differs from upstream—it causes issues for laypeople who don’t understand what’s going on as the system behaves bizarrely (in their eyes). 

If we want to be cautious, I think the best route is to state that SELinux needs to be disabled/permissive to run Lemmy via these playbooks. 

I’ll look today to see how much tweaking is needed to get these playbooks & Lemmy running with SELinux enabled/enforcing. Still, I think we’ll see more issues in the long run with all of the variables SELinux throws in that we would need to account for. If we want to go this route, I don’t mind supporting it. It will likely require manual work from the server operator/proper documentation (which might be beneficial to toss in the official Lemmy docs for RHEL). 

I’ve reiterated this point—these playbooks should be as frictionless as possible. In its current state, it is not possible to get Lemmy going/installed without modifications to SELinux, which will open a whole new can of worms.

Reply:
I agree with keeping them as frictionless as possible.

Reply:
Alright if its that much effort then lets keep it simple.